src/qt/clientmodel.h

```diff
@@ -79,6 +81,9 @@ class ClientModel : public QObject
     mutable std::atomic<int64_t> cachedBestHeaderTime;
     mutable std::atomic<int> cachedNumBlocks;
 
+    mutable std::atomic<uint256> cachedBestHeaderHash;
+    mutable std::atomic<uint256> cachedBestBlockHash;
```

Comment:
In commit "Added best block hash to the NotifyHeaderTip and NotifyBlockTip signals." (4948d4ee50d9960b23f43e14938845b0bb21772b)

These lines don't seem to work with my compiler version (clang version 6.0.0-1ubuntu2). I don't think it's possible to use std::atomic with the `uint256` class because it will only work for primitive int and pointer types. You probably need something like:

```c++
Mutex m_cached_tip_mutex;
uint256 m_cached_tip_headers;
uint256 m_cached_tip_blocks;
```

Reply:
done